src/selectors/name-generator.js

```diff
@@ -38,38 +38,30 @@ export const makeGetPromptNodeAttributes = () => {
   );
 };
 
-export const getPrimaryDisplay = createSelector(
-  propNodeDisplay,
-  nodeDisplay => nodeDisplay.primary,
+export const getCardDisplayLabel = createSelector(
+  propCardOptions,
+  cardOptions => cardOptions.displayLabel,
 );
 
-const propSecondaryDisplay = createSelector(
-  propNodeDisplay,
-  nodeDisplay => (has(nodeDisplay, 'secondary') ? nodeDisplay.secondary : []),
-);
-
-export const getSecondaryDisplay = createSelector(
-  propSecondaryDisplay,
-  secondary => secondary.map(property => omit(property, 'sortable')),
+export const getCardAdditionalProperties = createSelector(
+  propCardOptions,
+  cardOptions => (has(cardOptions, 'additionalProperties') ? cardOptions.additionalProperties : []),
 );
 
 export const getSortFields = createSelector(
-  getPrimaryDisplay,
-  propSecondaryDisplay,
-  (primary, secondary) => [primary].concat(secondary.filter(
-    property => property.sortable && property.variable).map(property => property.variable),
-  ),
+  propSortOptions,
+  sortOptions => (has(sortOptions, 'sortableProperties') ? sortOptions.sortableProperties : []),
 );
 
 export const getSortOrderDefault = createSelector(
-  getSortDefault,
-  sortData => sortData && (Object.keys(sortData)[0] || ''),
+  propSortOptions,
+  sortOptions => sortOptions.sortOrder && (Object.keys(sortOptions.sortOrder)[0] || ''),
```

Comment:
This isn't a comment on this PR, since it existed already, but is the intent to only allow a single sort field? If so, should we make that explicit with something like "sortField" and "sortDirection"? If not, should we use an array?

I imagine key ordering is implementation-specific.

Reply:
If there is only one then key ordering wouldn't be a problem. ;) But yes, at least I can't imagine a scenario where more than one would make sense, so maybe it should be explicit in the protocol.

Reply:
I can see that the API as it is now doesn't really lend itself to supporting multiple sort properties, although this would be a likely candidate for a feature request in the future. 

I think the reason we did it this way was just brevity, but I agree that it does introduce readability problems. Since it is consistent throughout the app right now, let's leave it for a separate issue.